### PR TITLE
Add a new API sample

### PR DESCRIPTION
**Pull request by mhegazy**
Add my sample in https://github.com/Microsoft/TypeScript/issues/5550#issuecomment-154773233 to the using the compiler API page.

### CODE REVIEW

File: Using-the-Compiler-API.md [hunk 1]
```diff
@@ -433,4 +433,154 @@ function compile(sourceFiles: string[], moduleSearchLocations: string[]): void {
     
     /// do something with program...
 }
-```
\ No newline at end of file
+```
+
+### Using the TypeChecker
```
Comment (DanielRosenwasser) [inline Using-the-Compiler-API.md]: Type Checker

File: Using-the-Compiler-API.md [hunk 2]
```diff
@@ -433,4 +433,154 @@ function compile(sourceFiles: string[], moduleSearchLocations: string[]): void {
...
+
+In this sample we will walk the AST and use the checker to serialize class information.
+We use the typeCheker to get symbol and type information, including JSDoc comments for the class, constructor and constructor parameters.
```
Comment (DanielRosenwasser) [inline Using-the-Compiler-API.md]: type checker

File: Using-the-Compiler-API.md [hunk 3]
```diff
@@ -433,4 +433,154 @@ function compile(sourceFiles: string[], moduleSearchLocations: string[]): void {
...
+        }
+
+        if (node.kind === ts.SyntaxKind.ClassDeclaration) {
+            // This is a class, get its symbol
+            let symbol = checker.getSymbolAtLocation((<ts.ClassDeclaration>node).name);
+            output.push(serializeClass(symbol));
+        }
+        else if (node.kind === ts.SyntaxKind.ModuleDeclaration) {
+            // This is a namespace, visit its children
+            ts.forEachChild(node, visit);
```
Comment (DanielRosenwasser) [inline Using-the-Compiler-API.md]: Class declarations can occur anywhere, you might as well just always `forEachChild`

File: Using-the-Compiler-API.md [hunk 4]
```diff
@@ -433,4 +433,154 @@ function compile(sourceFiles: string[], moduleSearchLocations: string[]): void {
...
+            documentation: ts.displayPartsToString(signature.getDocumentationComment())
+        };
+    }
+
+    /** True if this is visible outside this file, false otherwise */
+    function isNodeExported(node: ts.Node): boolean {
+        return (node.flags & ts.NodeFlags.Export) !== 0 || (node.parent && node.parent.kind === ts.SyntaxKind.SourceFile);
+    }
+
+    /** vist nodes finding exported classes */
```
Comment (DanielRosenwasser) [inline Using-the-Compiler-API.md]: Visit

File: Using-the-Compiler-API.md [hunk 5]
```diff
@@ -433,4 +433,154 @@ function compile(sourceFiles: string[], moduleSearchLocations: string[]): void {
...
+        };
+    }
+
+    /** True if this is visible outside this file, false otherwise */
+    function isNodeExported(node: ts.Node): boolean {
+        return (node.flags & ts.NodeFlags.Export) !== 0 || (node.parent && node.parent.kind === ts.SyntaxKind.SourceFile);
+    }
+
+    /** vist nodes finding exported classes */    
+    function visit(node: ts.Node) {
```
Comment (DanielRosenwasser) [inline Using-the-Compiler-API.md]: This is so far down compared to the actual call to `visit`. Order these such that they reflect when the calls occur, from earlier on top, to later on the bottom.

File: Using-the-Compiler-API.md [hunk 6]
```diff
@@ -433,4 +433,154 @@ function compile(sourceFiles: string[], moduleSearchLocations: string[]): void {
...
     /// do something with program...
 }
-```
\ No newline at end of file
+```
+
+### Using the Type Checker
+
+In this example we will walk the AST and use the checker to serialize class information.
+We'll use the type checker to get symbol and type information, while grabbing JSDoc comments for classes, their constructors, and respective constructor parameters.
```
Comment (DanielRosenwasser) [inline Using-the-Compiler-API.md]: for "exported" classes

**Comment by mhegazy** [comment]
@DanielRosenwasser, any more comments?

**Comment by DanielRosenwasser** [comment]
Nice! As long as the sample still works, :+1:

**Pull request merged**